operator/controllers/actionsrunner/util/converters.go

```diff
@@ -375,6 +317,109 @@ func ToJob(actionsRunner *inlocov1alpha1.ActionsRunner, actionsRunnerJob *inloco
 	return &job, nil
 }
 
+func withVolumes(actionsRunner *inlocov1alpha1.ActionsRunner) []corev1.Volume {
+	volumeByName := map[string]corev1.Volume{}
```

Comment:
As this map is only indexing the values of an existing slice, consider changing its value type to a pointer to avoid copying the values multiple times. This would require changing the statements that access this map as well.

Additionally, you would be able to avoid `realloc`-like calls by creating the map with its initial size set to the number of elements you are indexing.

```suggestion
	volumeByName := make(map[string]*corev1.Volume, len(actionsRunner.Spec.Volumes))
```